fx2eeprom.c

```diff
@@ -44,31 +44,37 @@
 #define EEPROM 0xa2
 #define TIMEOUT 1000
 #define LOCATION (1<<4)
+#define CHUNKSIZE 64
 
 enum {READ,WRITE};
 
 void use(char *prog){
-	fprintf(stdout,"%s w/r VID PID [size]\n",prog);
+	fprintf(stdout,"%s w/r VID PID [ size [ee_addr] ]\n",prog);
 	return;
 }
 
 int main(int argc, char *argv[]){
 	struct libusb_device_handle *dev;
 	int ret;
-	unsigned char *buffer;
-	int pid, vid, length;
+	unsigned char buffer[CHUNKSIZE];
```

Comment:
I prefer to keep the memory allocation, it simplifies the code.

Reply:
Are you able to work with files > 4K bytes? I got failures when I did so, while reading/writing chunkwise worked also with 16K byte (as well as 64K where I got 4x mirrored data).